### PR TITLE
Stop running new jobs in CI as soon as one of them fails

### PR DESCRIPTION
**Pull request by Ao-senXiong**
Fail-fast syntax is here https://docs.github.com/en/actions/writing-workflows/workflow-syntax-for-github-actions#jobsjob_idstrategyfail-fast

### CODE REVIEW

File: .github/workflows/ci.yml [hunk 1]
```diff
@@ -62,6 +62,7 @@ jobs:
     permissions:
       contents: read
     strategy:
+      fail-fast: true
```
Comment (wmdietl) [inline .github/workflows/ci.yml:65]: For `otherjdks` below we have `fail-fast: false`.
Is it possible for `remainder` to only cancel `otherjdks`?
I think it is good to have all remainder tasks run, even if one of them fails, to be able to see whether just that one task fails or maybe others fail, too.
But if any of the `remainder` tasks fails, we don't need to run any `otherjdks` tasks - one should first fix the failures revealed in JDK 21.
Does that policy make sense? Can it be expressed here?

Comment (Ao-senXiong) [reply .github/workflows/ci.yml:65]: The policy sounds good to me. Let me think about how to implement it.

Comment (Ao-senXiong) [reply .github/workflows/ci.yml:65]: @wmdietl I have looked around online and github workflow syntax. I did not find a nice solution for this policy: one job depends on another job's running status and both jobs are running parallel. Do you have any idea how to achieve this purpose?

Comment (Ao-senXiong) [reply .github/workflows/ci.yml:65]: make everything fail fast.

**Comment by Ao-senXiong** [comment]
Ideally, it is good to have all remainder tasks run, even if one of them fails, to be able to see whether just that one task fails or maybe others fail, too. But if any of the remainder tasks fails, we don't need to run any otherjdks tasks - one should first fix the failures revealed in JDK 21. From https://github.com/eisop/checker-framework/pull/894#discussion_r1736681845

**Review by wmdietl — APPROVED** [review]
Thanks for looking into alternatives!

**Pull request merged**